File: fibonacci/cxx/07_negative/fibonacci.cpp
#include <iostream>
#include <string>
#include "fibonacci.hpp"

int main(int argc, char** argv) {
  if (argc != 2) {
    std::cerr << "Error: " << argv[0] << " N" << std::endl;
    return 1;
  }
  std::cout << fibonacci(std::atoi(argv[1])) << std::endl;
  return 0;
}

File: fibonacci/julia/07_negative/fibonacci.jl
function fibonacci(n)
    if n < 0
        return fibonacci(n + 2) - fibonacci(n + 1)
    end
    if n == 0
        return 0
    end
    if n == 1
        return 1
    end
    return fibonacci(n - 1) + fibonacci(n - 2)
end
